modify conda python RPATH to solve openssl incompatibility

mmiesch opened the pull request:
## Description

Together with https://github.com/JCSDA-internal/ioda/pull/500, this seems to solve https://github.com/jcsda-internal/ioda/issues/463 in the gnu-openmpi-dev container.

More specifically, the `ctest -R python` tests currently fail for the develop branch of ioda in the gnu-openmpi-dev singularity container.  This is due to a conflict between the ioda python bindings and the libraries installed by conda, particularly openssl.  

Note: this is only a problem in the gnu-openmpi-dev singularity container.  Miniconda is not installed in the docker containers used for CI and it is not installed in the clang and intel singularity containers.  It is only installed in the gnu container because that is used for several of the [online tutorials](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/learning/tutorials/index.html), which require python packages such as cartopy.

With this PR, the ioda python tests now pass in the singularity container.

If you would like to test it, a copy of the new container is available here:

```
aws s3 cp s3://privatecontainers/jedi-gnu-openmpi-dev_beta.sif .
```

Note that this container is not signed and I cannot push it to our repository on Sylabs cloud because @mer-a-o now manages that repository.  When she returns from PTO we can update the "official" `library::jcsda/public/gnu-openmpi-dev:latest` container.

### Issue(s) addressed

Link the issues to be closed with this PR
- partiall-fixes #https://github.com/jcsda-internal/ioda/issues/463


## Acceptance Criteria (Definition of Done)

All tests for the develop branch of ioda-bundle pass in the container:

```
100% tests passed, 0 tests failed out of 448
```

## Dependencies

None

## Impact

None

Commit:
modify conda python RPATH to solve openssl incompatibility

srherbener commented:
> never mind - it works if I add `<build>/lib/python3.9/pyioda` to the `PYTHONPATH` in addition to `<build>/lib/python3.9`

Sorry about that - this was a change I pushed to the ioda PR just recently (to prevent the creation of two instances of the _ioda_python library).

mmiesch commented:
> > never mind - it works if I add `<build>/lib/python3.9/pyioda` to the `PYTHONPATH` in addition to `<build>/lib/python3.9`
> 
> Sorry about that - this was a change I pushed to the ioda PR just recently (to prevent the creation of two instances of the _ioda_python library).

So, does this mean we should update the [public wiki](https://github.com/JCSDA/fv3-bundle/wiki) with new instructions for the ioda python api?   Also, when I finish fine-tuning and testing the new container I want to write up something for jedi-docs.  At first I thought it might go in the FAQ but now I'm thinking there maybe should be a section in the ioda part of the docs on installing the python bindings and dealing with conda conflicts?

Commit:
specify LD_LIBRARY_PATH

Commit:
add pyyaml for ioda-converters

Commit:
re-install bufr after conda

mmiesch commented:
Thanks Steve - [I thought I had added that to the container](https://github.com/JCSDA-internal/containers/blob/2c8b052e313822a377880e65db226806877ee5e6/Singularity.gnu-openmpi-dev#L17) but maybe I forgot to rebuild it after I added that.   So that was only for the `ncepbufr` test, right?  Anyway, I'll check why it wasn't pre-set.

Good to know it works but we should think about this PR before we merge it.  I'm not sure it's an improvement if the "business as usual" approach of just cloning a bundle and building fails and/or gives all those warnings.

srherbener reviewed (approved):
@mmiesch thanks for coming up with a solution!

I'm not crazy about the patchelf step on the miniconda python executable because I think this makes the miniconda installation somewhat fragile. However, it works for all the ioda and ioda-converter tests and the issues I think would tend to only come up if someone started loading more python modules in their user space (and perhaps this is not common yet).

I think for pragmatic reasons it would be good to approve this and get it merged in, but we should also try to get the long term solution into place relatively soon to avoid issues with common libraries between python and jedi-stack.

mmiesch commented:
In light of our discussion this morning, I'm going to close this and try another approach of eliminating conda from the container.
